### DISCUSSION TITLE
Housing APIs

### DISCUSSION THREAD

**Discussion by Semisol** [HypixelDev/PublicAPI#355] [Ideas]
(Extending from #314)
I would like to see some APIs which can help me do some stuff that I am currently working on:
/player endpoint gets a field for house lists, name and UUID.
~~A latest given cookies object in the /player endpoint~~ Already done with `housingMeta.given_cookies_xxxx` I think
/house?id=house_id endpoint that returns some info about a house, such as owner and cookies.

**Comment by ConnorLinfoot** [comment]
Similar to the linked issue this is unlikely something we will add as it would be used very little and not worth the time implementing an API.

**Reply by Semisol** [reply]
> Similar to the linked issue this is unlikely something we will add as it would be used very little and not worth the time implementing an API.

Any way that we can get the ID of a house?

**Reply by MartinNemi03** [reply]
> 
> 
> Similar to the linked issue this is unlikely something we will add as it would be used very little and not worth the time implementing an API.

It's understandable, that it potentially wouldn't be as used as you expect it to, but as you expand the housing infrastructure and it's options, I think it could be at least attempted, since if you provide a good enough API endpoint, it can have it's potential in making Housing websites, tools and Discord bots a thing.

**Reply by Semisol** [reply]
Yep.

**Reply by Semisol** [reply]
And also ID is available via /wtfmap

**Reply by MartinNemi03** [reply]
Those IDs could also be potentially used to join Housings via a simple command, so that it could be possible for sites to provide an option to copy house's ID and play it. Like Fortnite's Creative island codes. But maybe I'm going too far in here, when we're talking just about the API.

**Reply by Semisol** [reply]
> Those IDs could also be potentially used to join Housings via a simple command, so that it could be possible for sites to provide an option to copy house's ID and play it. Like Fortnite's Creative island codes.

Interesting idea. Or some way to join by redirecting to the Hypixel site, where you can select to join or not, and it will warp you in game if you are on (forum account).

**Reply by Semisol** [reply]
I am looking forward to this expanding but the limitations are making me move to other ways to host small minigames.

**Reply by MartinNemi03** [reply]
Time to introduce Housing scripting with Lua. :D

**Reply by Semisol** [reply]
> Time to introduce Housing scripting with Lua. :D

Java + Bukkit API?
*starts panicking in sandboxing problems*

**Reply by TreasonMay** [reply]
Parkour times would also be a nice metric to have available via the API.

**Reply by TreasonMay** [reply]
Ranks would also be nice.
Maybe even giving action plates the ability to perform Webhooks?

**Reply by TreasonMay** [reply]
> > Those IDs could also be potentially used to join Housings via a simple command, so that it could be possible for sites to provide an option to copy house's ID and play it. Like Fortnite's Creative island codes.
> 
> Interesting idea. Or some way to join by redirecting to the Hypixel site, where you can select to join or not, and it will warp you in game if you are on (forum account).

I think a system similar to what Minecraft: Education Edition uses would be nice. (https://education.minecraft.net/wp-content/uploads/Minecraft-Education-Edition-Multiplayer-Guide-1.pdf)

**Reply by Semisol** [reply]
> Ranks would also be nice.
> Maybe even giving action plates the ability to perform Webhooks?

w e b h o o k s

isnt that too much
cant we add sandboxed bukkit apis already if that gets added

**Reply by TreasonMay** [reply]
The minigames server Minr has a scripting language for map creators, something like that would be the way to go I think.
